src/main.rs

```diff
@@ -286,34 +282,45 @@ fn main() -> Result<i32> {
         ),
     };
     let pretty = args.pretty.unwrap_or_else(|| buffer.guess_pretty());
-    let mut printer = Printer::new(pretty, args.style, args.stream, buffer);
+    let mut printer = Printer::new(print, pretty, args.style, args.stream, buffer);
 
-    if print.request_headers {
-        printer.print_request_headers(&request)?;
-    }
-    if print.request_body {
-        printer.print_request_body(&mut request)?;
-    }
+    printer.print_request_headers(&request)?;
+    printer.print_request_body(&mut request)?;
 
     let mut exit_code: i32 = 0;
     if !args.offline {
-        let response = client.execute(request)?;
-        let status = response.status();
-
-        exit_code = match status.as_u16() {
-            _ if !(args.check_status || args.download) => 0,
-            300..=399 if !args.follow => 3,
-            400..=499 => 4,
-            500..=599 => 5,
-            _ => 0,
+        let response = if args.follow {
+            let mut client =
+                redirect::RedirectFollower::new(&client, args.max_redirects.unwrap_or(10));
+            if args.all {
+                client.on_redirect(|prev_response, next_request| {
+                    printer.print_response_headers(&prev_response)?;
+                    printer.print_response_body(prev_response)?;
+                    printer.print_seperator()?;
+                    printer.print_request_headers(next_request)?;
+                    printer.print_request_body(next_request)?;
+                    Ok(())
+                });
+            }
+            client.execute(request)?
+        } else {
+            client.execute(request)?
         };
+
+        let status = response.status();
+        if args.check_status {
+            exit_code = match status.as_u16() {
+                300..=399 if !args.follow => 3,
+                400..=499 => 4,
+                500..=599 => 5,
+                _ => 0,
+            }
+        }
         if is_redirect && exit_code != 0 {
```

Comment:
`is_redirect` is now a bit ambiguous and could be confused with HTTP redirect. Maybe `is_redirected_output` or a similar thing would help?